src/sentry/web/forms/accounts.py

```diff
@@ -498,10 +531,27 @@ def __init__(self, project, user, *args, **kwargs):
         has_alerts = project.is_user_subscribed_to_mail_alerts(user)
         has_workflow = project.is_user_subscribed_to_workflow(user)
 
+        # This allows users who have entered an alert_email value or have specified an email
+        # for notifications to keep their settings
+        alert_email = UserOption.objects.get_value(user=self.user, project=None, key='alert_email', default=user.email,) or user.email
+        alert_email_choice = (alert_email, alert_email)
+        specified_email = UserOption.objects.get_value(user, project, 'mail:email', None)
+        specified_email_choice = (specified_email, specified_email)
+        choices = [(email.email, email.email) for email in user.get_verified_emails()]
+
+        if (specified_email_choice not in choices) and (specified_email_choice != (None, None)):
+            choices.append(specified_email_choice)
+        if (alert_email_choice not in choices) and (alert_email_choice != (None, None)):
+            choices.append(alert_email_choice)
+        if (user.email, user.email) not in choices:
+            choices.append((user.email, user.email))
+
+        self.fields['email'].choices = choices
```

Comment:
this seems like it could be simplified a bit. what if you did something like:

```python
emails = [e.email for e in user.get_verified_emails()]
# i think you can skip the default here if you're going to add the user.email below anyway
alert_email = UserOption.objects.get_value(user=self.user, project=None, key='alert_email', default=None)
specified_email = UserOption.objects.get_value(user, project, 'mail:email', None)
emails.extend([user.email, alert_email, specified_email])

choices = [(email, email) for email in set(emails) if email is not None]
self.fields['email'].choices = choices
```

also definitely keep your comment because it's helpful to know it's for supporting legacy email settings